core/tests/ts-integration/tests/fees.test.ts

```diff
@@ -213,6 +247,107 @@ testFees('Test fees', function () {
         console.log(`Full report: \n\n${reports.join('\n\n')}`);
     });
 
+    test('Test gas price expected value', async () => {
+        const receiver = ethers.Wallet.createRandom().address;
+        const l1GasPrice = 2_000_000_000n; /// set to 2 gwei
+        await mainNodeSpawner.killAndSpawnMainNode({
+            newL1GasPrice: l1GasPrice,
+            newPubdataPrice: l1GasPrice
+        });
+
+        const doTxToReceiver = () =>
+            alice.sendTransaction({
+                to: receiver,
+                value: BigInt(1)
+            });
+
+        //wait 1 miniblock so the multiplier gets updated
+        await doTxToReceiver().then((tx) => tx.wait(1));
```

Comment:
Multiplier is not guaranteed to be updated after each miniblock, it's updated async. Do I miss something? Does it make sense to poll L1 and wait for multiplier to be updated?

Reply:
This is a bad comment, it should be `wait 1 miniblock so the l1 gas price gets updated`

Reply:
Though probably polling is also better.

Reply:
Ok, so after drafting a wait like this:

```
 let gasPriceGood = false
        for (let i = 0; i < 40; i++) {
            await sleep(0.25);
            const l1GasPrice = (await alice._providerL2().getFeeParams()).V2.l1_gas_price
            if (l1GasPrice === l1GasPrice) {
                gasPriceGood = true;
                break;
            }
            console.log(l1GasPrice)
        }
        if (!gasPriceGood)
            throw new Error("Gas price change doesn't work");
```

I'm having the test fail due to unexpected gasPrice. 

```
    expect(received).toBe(expected) // Object.is equality

    Expected: 33912000n
    Received: 43960000n

      304 |         console.log('gas price', await alice._providerL2().getGasPrice());
      305 |         console.log(await alice._providerL2().getFeeParams());
    > 306 |         expect(receipt.gasPrice).toBe(BigInt(expectedConvertedGasPrice));
          |                                  ^
      307 |     });
      308 |
      309 |     test('Test base token ratio fluctuations', async () => {

      at Object.<anonymous> (tests/fees.test.ts:306:34)
```

This means fee params checked just after the transaction do not match what was used for gasPrice for the transaction. I'm not sure what exactly is causing it / what should we wait for, but I have seen consistently that doing one transaction and waiting for it to be confirmed before doing the tests transaction results in consistent results.

Reply:
So I want to just update the comment. Unless you have some idea what I can do better here

```suggestion
        //wait 1 miniblock so the gas price gets updated
        await doTxToReceiver().then((tx) => tx.wait(1));
```

Reply:
fee params are updated each batch, not miniblock. I'm ok with waiting for batch (please reuse existing function to do that) and changing comment

Reply:
Got it! Fixed.